Source/lighting.cpp

```diff
@@ -578,6 +579,8 @@ void ProcessLightList()
 			i--;
 			continue;
 		}
+		if (TileHasAny(dPiece[light.position.tile.x][light.position.tile.y], TileProperties::Solid))
+			continue; // Monster hidden in a wall, don't spoil the surprise

```

Comment:
Do we need to update `dLight` when the tile gets visible again?
(For example setting `Light::hasChanged` and `UpdateLighting`)

Reply:
All lighting should be invalidated when the map changes, which should take care of this.

Reply:
Out of curiosity I tested it (because I couldn't find the corresponding sources).
I disabled `AddObjectLight` (removed `DoUnLight`) and lights from players.
This is the result after teleporting to Lazarus:
![grafik](https://user-images.githubusercontent.com/25415264/234035066-d7e4fd4b-9f71-4285-9388-f2c619472ada.png)
The lighting is not updated.
When `AddObjectLight` works normally, this is irrelevant, because the other light sources ensure lighting is recalculated for Lazarus.
So feel free to ignore this comment, if you want. It should be only relevant for modders.

Reply:
I don't think we can protect people from deleting important parts of the code :)